sdk/containerregistry/azure-containers-containerregistry/src/test/java/com/azure/containers/containerregistry/ContainerRepositoryAsyncClientUpdateIntegrationTests.java

```diff
@@ -0,0 +1,241 @@
+import com.azure.core.test.TestMode;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import reactor.core.publisher.Mono;
+import reactor.test.StepVerifier;
+
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.azure.containers.containerregistry.TestUtils.DIGEST_UNKNOWN;
+import static com.azure.containers.containerregistry.TestUtils.DISPLAY_NAME_WITH_ARGUMENTS;
+import static com.azure.containers.containerregistry.TestUtils.HELLO_WORLD_SEATTLE_REPOSITORY_NAME;
+import static com.azure.containers.containerregistry.TestUtils.TAG_TO_DELETE;
+import static com.azure.containers.containerregistry.TestUtils.TAG_TO_UPDATE;
+import static com.azure.containers.containerregistry.TestUtils.TAG_UNKNOWN;
+
+@Execution(ExecutionMode.SAME_THREAD)
+public class ContainerRepositoryAsyncClientUpdateIntegrationTests extends ContainerRegistryClientsTestBase {
+
+    private ContainerRepositoryAsyncClient client;
+    private String digest;
+    private boolean resetTag;
+    private boolean resetRepository;
+
+    private ContainerRepositoryAsyncClient getContainerRepositoryAsyncClient(HttpClient httpClient) {
+        return getContainerRepositoryBuilder(HELLO_WORLD_SEATTLE_REPOSITORY_NAME, httpClient).buildAsyncClient();
+    }
+
+    private ContainerRepositoryAsyncClient getUnknownContainerRepositoryAsyncClient(HttpClient httpClient) {
+        return getContainerRepositoryBuilder(TAG_UNKNOWN, httpClient).buildAsyncClient();
+    }
+
+    private ContainerRegistryAsyncClient getContainerRegistryAsyncClient(HttpClient httpClient) {
+        return getContainerRegistryBuilder(httpClient).buildAsyncClient();
+    }
+
+
+    @BeforeAll
+    static void beforeAll() {
+        StepVerifier.setDefaultTimeout(Duration.ofMinutes(30));
+    }
+
+    @BeforeEach
+    public void setup() {
+        resetTag = false;
+        digest = null;
+        importImage(HELLO_WORLD_SEATTLE_REPOSITORY_NAME, Arrays.asList(TAG_TO_UPDATE, TAG_TO_DELETE));
+
+    }
+
+    @AfterEach
+    public void cleanup() {
+        if (getTestMode() == TestMode.PLAYBACK) {
+            return;
+        }
+
+        ContainerRepositoryAsyncClient client = getContainerRepositoryAsyncClient(new NettyAsyncHttpClientBuilder().build());
+
+        if (digest != null) {
+            client.setManifestProperties(digest, defaultProperties).block();
+        }
+
+        if (resetTag) {
+            client.setTagProperties(TAG_TO_UPDATE, defaultProperties).block();
+        }
+
+        if (resetRepository) {
+            client.setProperties(defaultProperties).block();
+        }
```

Comment:
Do these methods need to run sequentially? If they can be ran in parallel I believe [Mono.when](https://projectreactor.io/docs/core/release/api/reactor/core/publisher/Mono.html#when-org.reactivestreams.Publisher...-) and blocking that would be more appropriate.